From 67e4b4ceaefea83a1e5c77a7760fa1f9b37589f4 Mon Sep 17 00:00:00 2001
From: Daniel Mihai <dmihai@microsoft.com>
Date: Thu, 1 Sep 2022 15:07:16 +0000
Subject: [PATCH 09/10] Merged PR 9805: Add support for MSHV

Cloud Hypervisor is able to use either /dev/mshv or /dev/kvm.
---
 src/runtime/pkg/resourcecontrol/cgroups.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/runtime/pkg/resourcecontrol/cgroups.go b/src/runtime/pkg/resourcecontrol/cgroups.go
index 4210392d..d4608458 100644
--- a/src/runtime/pkg/resourcecontrol/cgroups.go
+++ b/src/runtime/pkg/resourcecontrol/cgroups.go
@@ -64,7 +64,8 @@ func sandboxDevices() []specs.LinuxDeviceCgroup {
 	// In order to run Virtual Machines and create virtqueues, hypervisors
 	// need access to certain character devices in the host, like kvm and vhost-net.
 	hypervisorDevices := []string{
-		"/dev/kvm",         // To run virtual machines
+		"/dev/kvm",         // To run virtual machines using KVM
+		"/dev/mshv",        // To run virtual machines using MSHV
 		"/dev/vhost-net",   // To create virtqueues
 		"/dev/vfio/vfio",   // To access VFIO devices
 		"/dev/vhost-vsock", // To interact with vsock if
-- 
2.17.1

